src/Jackalope/Version/VersionHistory.php

```diff
@@ -190,9 +203,13 @@ public function addVersionLabel($versionName, $label, $moveLabel)
      */
     public function removeVersionLabel($label)
     {
-        // @codeCoverageIgnoreStart
-        throw new NotImplementedException();
-        // @codeCoverageIgnoreEnd
+        if ($this->hasVersionLabel($label)) {
+            $version = $this->versionLabels[$label];
+            $this->objectManager->removeVersionLabel($version->getPath(), $label);
+            unset($this->versionLabels[$label]);
+        } else {
+            throw new VersionException("No label '$label'");
```

Comment:
lets fail early and move the `if` situation out of the if-else. so `if(!$this->hasVersionLabel..) {throw...}`